src/lib/input/input.scss

```diff
@@ -13,12 +13,13 @@
   // If background matches current background then so should the color for proper contrast
   color: currentColor;
 
-  // By default, <input> has a padding, border, outline and a default width.
+  // By default, <input> has a padding, border, outline, a default width and a default height.
   border: none;
   outline: none;
   padding: 0;
   margin: 0;
   width: 100%;
+  height: $mat-input-height;
```

Comment:
I don't think we want to do this, the input is supposed to adapt to the font size of its parent, and this will probably break it.

Reply:
Good point. What about a min-height? I can update the PR with that.

Reply:
I still think that might be problematic. It will prevent the element from shrinking when the `font-size` is smaller. You could try setting `height: $line-height * 1em` here: https://github.com/angular/material2/blob/master/src/lib/input/_input-theme.scss#L27

Reply:
Much better. I am testing right now and will update the PR in the next couple of minutes.

Reply:
I had a problem with the screenshot tests. I did a rebase with the master branch and it worked fine except that the other tests (especially e2e) are not working. Did I forget something? Or is it just a temporary instability in the master branch?

Reply:
Looking at the demo, it seems to cause an issue in safari where the bottom of the text is clipped off. I played around with it a little bit and it seemed like maybe setting the `line-height` of the `:first-line` will help, but it'll need to be tested across all browsers

Reply:
Sorry for the delay. I am traveling, but I am looking on it. I will update the PR as soon as possible.

Reply:
I finally returned. This is what I see in safari. Could you help me to replicate the text clipping?

![safari_image](https://user-images.githubusercontent.com/5031156/33531961-6e6674ea-d873-11e7-9637-cbe0c1e1356d.png)

Reply:
Sorry for the slow response, take a look at these 2 demos on Safari:
https://mmalerba-demo1.firebaseapp.com/input (master)
https://mmalerba-demo2.firebaseapp.com/input (master + your pr)

Notice for the first input (the disabled one) that the 'g' in 'Google' is slightly clipped on the bottom with your PR.

Reply:
Sorry for the delay, @mmalerba. I was trying the `line-height` approach, but I discovered that setting the border changes the baseline in Safari, so I tried another approach and I would like to know your opinion about.

Reply:
I would prefer to avoid using `border` if possible, because even though its transparent it still shows up solid in high contrast mode on windows. (We actually do use a border hack somewhere else in the form field CSS, but we've been trying to remove it: https://github.com/angular/material2/pull/6291)

Reply:
@mmalerba Sorry again for the delay. I had to find a device with Safari for testing. I think the current version of the PR will solve the problem.